src/view/screens/Settings/ExportCarDialog.tsx

```diff
@@ -18,20 +19,17 @@ export function ExportCarDialog({
   const {_} = useLingui()
   const t = useTheme()
   const {gtMobile} = useBreakpoints()
-  const {currentAccount} = useSession()
   const {getAgent} = useAgent()
 
-  const downloadUrl = React.useMemo(() => {
+  const download = React.useCallback(async () => {
     const agent = getAgent()
-    if (!currentAccount || !agent.session) {
-      return '' // shouldnt ever happen
+    if (!agent.session) {
+      return // shouldnt ever happen
     }
-    // eg: https://bsky.social/xrpc/com.atproto.sync.getRepo?did=did:plc:ewvi7nxzyoun6zhxrhs64oiz
-    const url = new URL(agent.pdsUrl || agent.service)
-    url.pathname = '/xrpc/com.atproto.sync.getRepo'
-    url.searchParams.set('did', agent.session.did)
-    return url.toString()
-  }, [currentAccount, getAgent])
+    const did = agent.session.did
+    const res = await agent.com.atproto.sync.getRepo({did})
+    await saveBytesToDisk('repo.car', res.data, res.headers['content-type'])
```

Comment:
At least on a simulator the problem I've been having with this (or really, the underlying `Share`) is that it keeps opening in mail on Android with this mime type (I think it's ending up as `application/x-octet-stream`). I tried just removing the mime altogether worked to give me the full list of options, but there isn't a "save" option.

With that said, could be that I'm on a simulator. Do regular devices have this same problem?

Reply:
I'll try on my native device.

Reply:
The android share menu does not allow to "save file" as ios does (some custom file manager apps do enable that but the default one does not).

So to actually share on device, I had to change the code for android. There is still a fallback to regular sharing through apps if the user denied access to disk for the save operation.

I tested using my own (Google Pixel 8 Pro) phone and a lot of apps *do* appear in the sharing menu.